File: selfstudy/python/temperatures_min.py
'''
task 143:
Temperatures - Min
You need to find the minimum value among daily temperatures in Celsius
'''

def lowest_temperatures(temps):
    return min(temps) if temps else 0

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], lowest_temperatures([10, 20, 30])),
        ([50, 60, 70], lowest_temperatures([50, 60, 70])),
        ([1, 2, 3], lowest_temperatures([1, 2, 3])),
        ([100, 200], lowest_temperatures([100, 200])),
        ([], lowest_temperatures([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = lowest_temperatures(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# works i think

File: selfstudy/golang/temperatures_min.go
/*
task 243:
Temperatures - Min
You need to find the minimum value among daily temperatures in Celsius
*/

package main

import (
	"fmt"
	"reflect"
)

func LowestTemperatures(temps []int) int {
	if len(temps) == 0 { return 0 }
	m := temps[0]
	for _, v := range temps {
		if v < m { m = v }
	}
	return m
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, LowestTemperatures([]int{10, 20, 30})},
		{[]int{50, 60, 70}, LowestTemperatures([]int{50, 60, 70})},
		{[]int{1, 2, 3}, LowestTemperatures([]int{1, 2, 3})},
		{[]int{100, 200}, LowestTemperatures([]int{100, 200})},
		{[]int{}, LowestTemperatures([]int{})},
	}
	
	for idx, t := range tests {
		res := LowestTemperatures(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// finally works
